File: app/src/main/java/com/nollpointer/dates/other/PreferencesUploadManager.java
package com.nollpointer.dates.other;

import android.content.Context;
import android.os.AsyncTask;

//TODO класс, который полностью отвечает за загрузку значений из памяти телефона. Работает асинхронно
public class PreferencesUploadManager {

    static interface OnResultListener{

    }


    public static void get(Context context, String TAG){

    }

    public static void getAsync(String TAG, OnResultListener listener){

    }

    public static void set(String TAG, String object){

    }


    public static void setAsync(String TAG, String object, OnResultListener listener){

    }


    private class AsyncSet extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute() {
        }

        @Override
        protected void onPostExecute(Void aVoid) {
        }

        @Override
        protected Void doInBackground(Void... voids) {

            return null;
        }
    }

    private class AsyncGet extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute() {
        }

        @Override
        protected void onPostExecute(Void aVoid) {
        }

        @Override
        protected Void doInBackground(Void... voids) {

            return null;
        }
    }

}

File: app/src/main/java/com/nollpointer/dates/other/PreferencesUploadManager.kt
package com.nollpointer.dates.other

import android.content.Context
import android.os.AsyncTask

//TODO класс, который полностью отвечает за загрузку значений из памяти телефона. Работает асинхронно
class PreferencesUploadManager {

    operator fun get(context: Context, TAG: String) {}

    fun getAsync(TAG: String, listener: ()->Unit) {}

    operator fun set(TAG: String, obj: String) {}

    fun setAsync(TAG: String, obj: String, listener: ()->Unit) {}

    private inner class AsyncSet : AsyncTask<Void, Void?, Void>() {

        protected override fun doInBackground(vararg voids: Void): Void? {
            return null
        }
    }

    private inner class AsyncGet : AsyncTask<Void, Void?, Void>() {

        protected override fun doInBackground(vararg voids: Void): Void? {
            return null
        }
    }
}